packages/@sanity/types/src/schema/definition/schemaDefinition.ts

```diff
@@ -0,0 +1,131 @@
+import {PreviewConfig} from '../preview'
+import {InitialValueProperty, SchemaValidationValue} from '../types'
+import {
+  ArrayDefinition,
+  BlockDefinition,
+  BooleanDefinition,
+  CrossDatasetReferenceDefinition,
+  DateDefinition,
+  DatetimeDefinition,
+  DocumentDefinition,
+  FileDefinition,
+  GeopointDefinition,
+  ImageDefinition,
+  NumberDefinition,
+  ObjectDefinition,
+  ReferenceDefinition,
+  SlugDefinition,
+  SpanDefinition,
+  StringDefinition,
+  TextDefinition,
+  UrlDefinition,
+} from './type'
+import {BaseDefinitionOptions} from './type/common'
+
+/**
+ * This type can be extended being a interface and not a type, via declaration merging
+ */
+export interface IntrinsicTypeDefinition {
+  array: ArrayDefinition
+  block: BlockDefinition
+  boolean: BooleanDefinition
+  date: DateDefinition
+  datetime: DatetimeDefinition
+  document: DocumentDefinition
+  file: FileDefinition
+  geopoint: GeopointDefinition
+  image: ImageDefinition
+  number: NumberDefinition
+  object: ObjectDefinition
+  reference: ReferenceDefinition
+  crossDatasetReference: CrossDatasetReferenceDefinition
+  slug: SlugDefinition
+  span: SpanDefinition
+  string: StringDefinition
+  text: TextDefinition
+  url: UrlDefinition
+}
+
+/**
+ * A union of all intrinsic types allowed natively in the schema.
+ */
+export type TypeName = IntrinsicTypeDefinition[keyof IntrinsicTypeDefinition]['type']
+
+/**
+ * Represents a Sanity schema type definition.
+ *
+ * It's recommend to use the `defineType` helper instead of this type by
+ * itself.
+ */
+export type TypeDefinition<TType extends TypeName = TypeName> =
+  | IntrinsicTypeDefinition[TypeName]
+  | TypeAliasDefinition<string, TType>
+
+/**
+ * Represents a Sanity schema type definition with an optional type parameter.
+ *
+ * It's recommend to use the `defineType` helper instead of this type by
+ * itself.
+ *
+ * @see defineType
+ */
+export type SchemaTypeDefinition<TType extends TypeName = TypeName> = TypeDefinition<TType>
```

Comment:
I'm struggling to understand the difference between `TypeDefinition` and `SchemaTypeDefinition`

Reply:
They seem to be exactly the same: I agree. Think this is a result of the code living seperatly before, but now they come next to each other and its obvious. Will fix.